### DISCUSSION TITLE
[Turborepo + Next 14 + Tailwind] What's the best way of handling SVGs and passing props/styles to them?

### DISCUSSION THREAD

**Discussion by mariosffx** [vercel/next.js#67870] [Help]
Hello.

I have started using Turbo Repo and NextJS and I would like to create some UI packages and some of them, use SVG Icons. 
Additionally I might use some of those icons in one my apps. 

Based on the above needs, I believe it would be best to have those icons in a package and import them wherever I need them.

The problem is that I got lost in tutorials, many guides and some projects use `svgr/webpack` which can convert  SVGs to React Components and then pass props to them. That's fine if I setup SVGR in a project/app and provide additional configuration in `next.config.mjs`.

However, I'm concerned that in a future versions of NextJS, webpack might not be used anymore and setting it up as a package is complicated as I receive errors that React doesn't know how to load SVGs and there might be better approaches of creating an Icon Library and that's why I seek help here.

My project setup uses the NextJS ^14  and it was built by running the `npx create-turbo@latest` command

**Comment by icyJoseph** [comment]
Go with SVGR.

- https://x.com/jaredpalmer/status/1619071988181651456?s=46&t=OYS-XorPj2maJAaJJdWahw
- https://github.com/gregberge/svgr/issues/816
